lib/trino-hdfs/src/main/java/io/trino/hdfs/HdfsConfigurationProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hdfs;
+
+import org.apache.hadoop.conf.Configuration;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+
+public class HdfsConfigurationProvider
+        implements DynamicConfigurationProvider
+{
+    private final boolean cachingDisabled;
+
+    @Inject
+    public HdfsConfigurationProvider(HdfsConfig hdfsConfig)
+    {
+        this.cachingDisabled = hdfsConfig.getFileSystemMaxCacheSize() == 0;
```

Comment:
What about negative values? I guess it causes the same error "FileSystem max cache size has been reached" We could add `@Min` annotation to `HdfsConfig#getFileSystemMaxCacheSize` if we want to disallow such values.